string action

---

RuleMsg[] rule_set
